### PR TITLE
Migrate to TS

### PR DESCRIPTION
**Pull request by johnnyasantoss**
- Upgrade dependencies
- Migrate to Typescript
- Use pnpm
- Add response and requests types

### CODE REVIEW

File: src/types/v2.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+export interface OpenNodePayoutTransferRequest {
+  /** Use type=email for internal transfers, type=managed to transfers to managed accounts */
+  type: "email" | "managed";
```
Comment (joaodealmeida) [inline src/types/v2.ts]: Should be "id" | "managed"

File: src/types/v2.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  /** Recipient's OpenNode ID or managed account ID */
+  address: string;
+  /** Amount in source wallet's currency - i.e. if wallet=USD, amount is in USD */
+  amount: string;
```
Comment (joaodealmeida) [inline src/types/v2.ts:7]: currency missing

File: src/types/v2.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  wallet: string;
+
+  transfers: OpenNodePayoutTransferRequest[];
+
+  /** URL to receive webhooks for payout status updates */
+  webhook_url?: string;
+}
+
+export interface OpenNodePayout {
+  id: string;
```
Comment (joaodealmeida) [inline src/types/v2.ts:25]: Outdated, new serialization

```
  id: string;
  amount: number;
  fee: number;
  source_amount: number;
  source_fee: number;
  source_wallet: string;
  status: PayoutStatus;
  transfers?: PayoutTransferSerialized[];
  external_funding: boolean;
  created_at: Date;
  processed_at: Date;
  ```

File: src/types/v2.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+export interface OpenNodeLnURLWithdrawalRequest {
+  min_amt: number;
+  max_amt: number;
+  callback_url?: string;
+  external_id?: string;
+  expiry_date?: string;
+  description?: string;
+}
```
Comment (joaodealmeida) [inline src/types/v2.ts:168]: should be a number (unix epoch)

File: src/types/v2.ts [hunk 5]
```diff
@@ -94,18 +153,25 @@ export interface OpenNodeWithdrawal {
...
   processed_at: number;
-  status: string;
+  conversion?: {
+    from: string;
+    from_amount: number;
+    to: string;
+    to_amount: number;
+  };
+  expiry?: number;
+  status: "completed" | "initial" | "pending" | "failed";
```
Comment (joaodealmeida) [inline src/types/v2.ts]: initial is not public facing

confirmed is also a state (legacy compatible), completed only used on "wires" public facing-responses

**Review by joaodealmeida — CHANGES_REQUESTED** [review]
Added comments

File: src/types/v2.ts (hunk 5 above)
Comment (johnnyasantoss) [reply src/types/v2.ts]: Will revert to string to accommodate all cases

**Review by joaodealmeida — APPROVED** [review]
LGTM

(merge on `develop` instead of `master` first)

**Review by joaodealmeida — CHANGES_REQUESTED** [review]
Tests are not running

```
> opennode@1.5.0 test
> mocha


Error: Cannot find module '../src/lib'
```

**Review by joaodealmeida — APPROVED** [review]

**Pull request merged**